src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -254,6 +254,31 @@ runtime::gpu::GPU_ExternalFunction::~GPU_ExternalFunction()
 {
 }
 
+void runtime::gpu::GPU_ExternalFunction::assemble_and_reserve()
+{
+    GPUAllocator allocator = m_shared_context->m_primitive_emitter->get_memory_allocator();
```

Comment:
may add this part to another`cache_ordered_list()`.

Reply:
Is line 271 kind of what you were meaning? Currently storing the buffers into cache with a lookup key based on the function name.
https://github.com/NervanaSystems/ngraph/pull/1231/files/74df2123da1095fe42544f0587fb19e405ff768a#diff-d68b46ecac06f2a6346a24a2af11f87aR271